membership-common/src/main/scala/com/gu/memsub/subsv2/Subscription.scala

```diff
@@ -76,7 +76,10 @@ object GetCurrentPlans {
       else if (product == Product.Digipack && plan.billingPeriod.toOption.contains(OneTimeChargeBillingPeriod))
         digipackGiftEnded(_ => sub.termEndDate >= dateToCheck)
       else
-        paidPlanEnded(_ => plan.effectiveEndDate >= dateToCheck)
+        paidPlanEnded(_ => {
+          val inGracePeriodAndNotCancelled = plan.effectiveEndDate == dateToCheck && !sub.isCancelled
```

Comment:
if it's only just run out at midnight, and the subscription is not cancelled, we assume it will auto renew later in the day (1 day grace period)